spaceship/lib/spaceship/client.rb

```diff
@@ -447,12 +441,10 @@ def send_shared_login_request(user, password)
         end
 
         response = request(:post) do |req|
-          req.url "https://idmsa.apple.com/appleauth/auth/signin"
-          req.body = data.to_json
-          req.headers['Content-Type'] = 'application/json'
-          req.headers['X-Requested-With'] = 'XMLHttpRequest'
-          req.headers['X-Apple-Widget-Key'] = self.itc_service_key
-          req.headers['Accept'] = 'application/json, text/javascript'
+          req.url "https://idmsa.apple.com/IDMSWebAuth/authenticate"
+          req.body = "appIdKey=891bd3417a7776362562d2197f89480a8547b108fd934911bcbea0110d07f757&appleId=#{user}&accountPassword=#{password}"
```

Comment:
How did we arrive on this magic appIdKey ❓

Reply:
Great question, I cleared my history and navigated to developer.apple.com
then I clicked on `account` which took me to the login page `https://idmsa.apple.com/IDMSWebAuth/login?appIdKey=891bd3417a7776362562d2197f89480a8547b108fd934911bcbea0110d07f757&path=%2Faccount%2F&rv=1`
I then tracked the `POST` request and that `appIdKey` was included.
I resent the request without the appIdKey and it failed, so I added it back in and the request succeeded.
My guess is that Apple is using SSO now, and each service they operate will have an "appIdKey" associated with it.

Reply:
We might be able to derive this `appIdKey` from somewhere, but I didn't find it yet.

Reply:
If you google for it, you will find plenty of references on other's people scripts :)

Reply:
If you get https://developer.apple.com/account/, you get a 302 to the login page you hardcoded.

Reply:
Does this mean we can’t use passwords that contain `=` or `&`?

Reply:
I assume we have to url encode the password

```ruby
require 'uri'
URI.encode(password)
```

Reply:
and the username as well

Reply:
Thank you, updated!

Reply:
it's possible to use
encoded_password = CGI.escape(password)
 encoded_apple_id = CGI.escape(user)

when you have "+" in user. it's not working